### PR TITLE
test: add core24 component spread tests

### PR DESCRIPTION
**Pull request by mr-cal**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox run -m lint`?
- [x] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

Flatten component spread tests and copy to `core24`.

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -283,6 +283,11 @@ suites:
    systems:
      - ubuntu-22.04*
 
+ tests/spread/core24/components/:
```
Comment (mr-cal) [inline spread.yaml:286]: I did this in less than 1 hour after @sergiusens said not to!

I'm not sure of an alternative way to get spread to recognize this directory.  The [spread docs](https://github.com/snapcore/spread?tab=readme-ov-file#the-cascading-matrix) state, "
_All tasks for a suite live under the same directory. One directory per suite, one directory per task, and task.yaml inside the latter._"

Comment (mr-cal) [reply spread.yaml:286]: Edit:  I think the alternative is to remove organized spread tests and make a flat layout:

From
```
tests/spread/
└── core22
    └── patchelf
        ├── classic-patchelf
        │   ├── snap
        │   └── task.yaml
        ├── classic-python
        │   └── task.yaml
        └── strict-patchelf
            ├── snap
            └── task.yaml
```

to

```
tests/spread/
└── core22
    ├── patchelf-classic
    │   ├── snap
    │   └── task.yaml
    ├── patchelf-classic-python
    │   └── task.yaml
    └── patchelf-strict
        ├── snap
        └── task.yaml
```

**Review by sergiusens — APPROVED** [review]

**Comment by sergiusens** [comment]
you can put a pin on this for now if you want and we can fix later

**Comment by mr-cal** [comment]
The work is done and CI is passing, so we may as well land it.  (it's on the `feature/components` branch anyways)

**Pull request merged**